gremlin/traversal/flows.go

```diff
@@ -703,6 +705,45 @@ func (r *RawPacketsTraversalStep) BPF(s ...interface{}) *RawPacketsTraversalStep
 	return &RawPacketsTraversalStep{GraphTraversal: r.GraphTraversal, rawPackets: rawPackets}
 }
 
+// SocketInfo returns the sockets at both sides of the specified flows
+func (f *FlowTraversalStep) Sockets(s ...interface{}) *SocketsTraversalStep {
+	if f.error != nil {
+		return &SocketsTraversalStep{error: f.error}
+	}
+
+	if len(s) != 0 {
+		return &SocketsTraversalStep{error: fmt.Errorf("Sockets requires no parameter")}
+	}
+
+	indexer := NewSocketIndexer(f.GraphTraversal.Graph)
+
+	sockets := make(map[string][]*socketinfo.ConnectionInfo)
+	for _, fl := range f.flowset.Flows {
+		if transport := fl.GetTransport(); transport != nil && transport.GetProtocol() == flow.FlowProtocol_TCPPORT {
+			if _, found := sockets[fl.UUID]; !found {
+				sockets[fl.UUID] = make([]*socketinfo.ConnectionInfo, 0)
```

Comment:
I could be nice to remove or to not set an empty entry if nothing match otherwise we get 

```
		"ba2ced311cb67b11679cccc7a24374658ff32db1": [],
		"e076a84f3161cd0cb974b09e11f266894d051c5d": [],
		"f50ef4fff0b9bc70a7b431847b4da773857323f2": []
```

Reply:
done